src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -38,36 +54,53 @@ public Object getBean(String name) throws BeanNotFoundException {
         }
     }
 
+    /**
+     * Request beans from the storage with specified parameters.
+     *
+     * @param name         bean name
+     * @param requiredType required class type
+     * @return bean object if its exist or else throw exception.
+     */
     @Nullable
     @Override
-    public Object getBean(String name, Class<Object> requiredType) throws BeanNotFoundException {
+    public Object getBean(@Nonnull final String name,
+                          @Nonnull final Class<Object> requiredType) throws BeanNotFoundException {
         Object bean = singletonBeans.get(name);
 
         if (bean == null) {
             throw new BeanNotFoundException("Bean with a name '" + name + "' not found.");
```

Comment:
Instead of using concatenation, it is better to use String.format() for substitutions and it has been optimised in latest JDK versions, therefore any significant impact on performance is not expected.